subnet/subnet_test.go

```diff
@@ -402,3 +403,129 @@ func TestWatchNetworkRemoved(t *testing.T) {
 		t.Errorf("WatchNetwork produced wrong network: expected %s, got %s", expected, actual)
 	}
 }
+
+func TestAddReservation(t *testing.T) {
+	msr := newDummyRegistry()
+	sm := NewMockManager(msr)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	r := Reservation{
+		Subnet:   newIP4Net("10.4.3.0", 24),
+		PublicIP: ip.MustParseIP4("52.195.12.13"),
+	}
+	if err := sm.AddReservation(ctx, "_", &r); err == nil {
+		t.Fatalf("added reservation outside of configured network")
```

Comment:
I feel this fatal msg is not clear enough. How about `unexpected adding reservation outside of configured network`